### PR TITLE
asserts: add snap-size and remove snap-build from snap-revision

### PR DESCRIPTION
**Pull request by emgee**
iirc, the snap-revision assertion was original tightly bound to the snap-build assertion by a combination of the snap-revision's header fields, one of which was called snap-build (really the digest of that assertion).

snap-revision's primary key was changed some time ago to uncouple it from the snap-build assertion. At that time, snap-revision's snap-build header became redundant and, more importantly, incorrect.

snap-size was added to snap-revision to give the assertion the same level of proof for the snap as snap-build has.

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by emgee** [comment]
Just noticed a related clean-up. Will push in a moment.

File: asserts/snap_asserts.go [hunk 1]
```diff
@@ -171,16 +172,16 @@ func (snaprev *SnapRevision) SnapDigest() string {
 	return snaprev.Header("snap-digest")
 }
 
+// SnapSize returns the size in bytes of the snap submitted to the store.
+func (snaprev *SnapRevision) SnapSize() uint64 {
+	return snaprev.snapSize
+}
+
 // SnapRevision returns the revision of the snap-id assigned to this build.
```
Comment (pedronis) [inline asserts/snap_asserts.go]: unrelated to this branch, but this comment is clunky/ambiguous, can we improve it?

SnapRevision returns the revision assigned to this build of snap-id ?

File: asserts/snap_asserts.go [hunk 2]
```diff
@@ -192,26 +193,15 @@ func (snaprev *SnapRevision) Timestamp() time.Time {
 	return snaprev.timestamp
 }
 
-// Implement further consistency checks.
```
Comment (pedronis) [inline asserts/snap_asserts.go]: we will still need this, probably to do checks vs SnapDeclaration

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: asserts/snap_asserts.go (hunk 2 above)
Comment (emgee) [inline asserts/snap_asserts.go]: Reinstated, but with no snap-build TODOs.

File: asserts/snap_asserts.go (hunk 1 above)
Comment (emgee) [inline asserts/snap_asserts.go]: Fixed. Also updated some doc comments that suggested a coupling with snap-build.

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by pedronis** [comment]
+1

File: asserts/snap_asserts.go [hunk 3]
```diff
@@ -152,10 +152,11 @@ func assembleSnapBuild(assert assertionBase) (Assertion, error) {
 }
 
 // SnapRevision holds a snap-revision assertion, which is a statement by the
-// store acknowledging the receipt of a snap build and labeling it with a snap
-// revision.
+// store acknowledging the receipt of a build of snap-id and labeling it with a
```
Comment (niemeyer) [inline asserts/snap_asserts.go]: The documentation was right before here. It was the snap that was built, not the snap-id.

File: asserts/snap_asserts.go [hunk 4]
```diff
@@ -171,18 +172,18 @@ func (snaprev *SnapRevision) SnapDigest() string {
...
-// SnapRevision returns the revision of the snap-id assigned to this build.
-func (snaprev *SnapRevision) SnapRevision() uint64 {
-	return snaprev.snapRevision
+// SnapSize returns the size in bytes of the snap submitted to the store.
+func (snaprev *SnapRevision) SnapSize() uint64 {
+	return snaprev.snapSize
 }
 
-// SnapBuild returns the digest of the associated snap-build.
-func (snaprev *SnapRevision) SnapBuild() string {
```
Comment (niemeyer) [inline asserts/snap_asserts.go]: Why is that being removed? That's one of the most critical reference in this assertion. Without it, what is the store claiming it received?

File: asserts/snap_asserts.go [hunk 5]
```diff
@@ -171,18 +172,18 @@ func (snaprev *SnapRevision) SnapDigest() string {
...
-	return snaprev.Header("snap-build")
+// SnapRevision returns the revision assigned to this build of snap-id.
+func (snaprev *SnapRevision) SnapRevision() uint64 {
+	return snaprev.snapRevision
 }
 
-// DeveloperID returns the id of the developer that submitted the snap build to
-// the store.
+// DeveloperID returns the id of the developer that submitted this build of
```
Comment (niemeyer) [inline asserts/snap_asserts.go]: Please preserve the original language here as well. This is about a snap build, not a snap-id build.

**Comment by niemeyer** [comment]
The snap size inclusion sounds fine, but the other changes don't seem sound.

File: asserts/snap_asserts.go (hunk 4 above)
Comment (niemeyer) [inline asserts/snap_asserts.go]: My apologies. As Samuele pointed out to me, the digest of the snap is still here, so my concern is sorted.

**Comment by niemeyer** [comment]
Given the last comment above, my core concern is sorted about this branch.

LGTM assuming the remaining trivial tweaks are in place.

**Comment by emgee** [comment]
Pushed doc comment tweaks. I went with "build of the snap" in place of "snap build" as both Samuele and I felt it was a little more explicit in its meaning. Hope that's OK!

**Pull request merged**